Unicode/memory.c

```diff
@@ -47,11 +47,10 @@ char *copyn(const char *str,long n) {
      *      https://github.com/fontforge/fontforge/issues/1239
      */
     char *ret;
-    if ( str==NULL )
-    	return( NULL );
-
-    ret = (char *) malloc(n+1);
-    memcpy(ret,str,n);
-    ret[n]='\0';
-    return( ret );
+    if ( str!=NULL && ++n>0 && (ret=(char *)(malloc(n)))!=NULL ) {
```

Comment:
This makes it quite obtuse in what it's trying to do, especially the stuff around incrementing then decrementing `n`.

Something like

```c
    char *ret;
    if ( str==NULL || n <=0 )   // <-- modify here
    	return( NULL );

    ret = (char *) malloc(n+1);
    if (ret == NULL)             // <-- add here
        return;
    memcpy(ret,str,n);
    ret[n]='\0';
return( ret );
```

Is way more readable. Although I'd question the necessity of the `<0` check on `n`, at that point you're just masking issues with code which is doing something bad.

Reply:
That is a lot more readable.